### PR TITLE
add --no-color option

### PR DESCRIPTION
**Pull request by nerdrew**
colorized output is messy when calling fast from my editor (vim).

### CODE REVIEW

File: lib/fast/cli.rb [hunk 1]
```diff
@@ -47,6 +49,7 @@ def initialize(args)
       args = replace_args_with_shortcut(args) if args.first&.start_with?('.')
 
       @pattern, *@files = args.reject { |arg| arg.start_with? '-' }
+      @colorize = STDOUT.isatty
```
Comment (jonatas) [inline lib/fast/cli.rb]: Today I learned ❤️

File: lib/fast/cli.rb [hunk 2]
```diff
@@ -14,14 +14,16 @@ module Fast
...
+  def highlight(node, show_sexp: false, colorize: true)
     output =
       if node.respond_to?(:loc) && !show_sexp
         node.loc.expression.source
       else
         node
       end
-    CodeRay.scan(output, :ruby).term
+
+    code = CodeRay.scan(output, :ruby)
```
Comment (jonatas) [inline lib/fast/cli.rb]: I never need to focus on performance, but I think we can skip processing the scanning part if we're not going to colorize and use `output` directly.
```suggestion
  return output unless colorize
```

**Comment by jonatas** [comment]
Thanks for the contribution @nerdrew! I'm curious if you can share any tips and tricks on how you use and integrate the tool in your editor. 

I also use vim but I still didn't integrate it very well. I use it more from a pure command line.

**Comment by nerdrew** [comment]
I use neovim + [neomake](https://github.com/neomake/neomake), but I think the basics apply even if you don't use neomake.

```vim
function! s:Fast(args)
  let cmd = ''
  if !empty(b:ruby_project_root)
    let cmd .= 'cd ' . b:ruby_project_root . ' && '
  endif

  let cmd .= 'fast --no-color ' . a:args

  let custom_maker = neomake#utils#MakerFromCommand(cmd)
  let custom_maker.name = cmd
  let custom_maker.cwd = b:ruby_project_root
  let custom_maker.remove_invalid_entries = 0
  " e.g.:
  "   # path/to/file.rb:1141
  "   my_method(
  "     :boom,
  "     arg1: 1,
  "   )
  " %W# %f:%l -> start a multiline warning when the line matches '# path/file.rb:1234'
  " %-Z# end multiline warning on the next line that starts with '#'
  " %C%m continued multiline warning message
  let custom_maker.errorformat = '%W# %f:%l, %-Z#, %C%m'
  let enabled_makers = [custom_maker]
  update | call neomake#Make(0, enabled_makers) | echom "running: " . cmd
endfunction
command! -complete=file -nargs=1 Fast call s:Fast(<q-args>)
```

https://github.com/nerdrew/dotvim/blob/master/ftplugin/ruby.vim#L140-L165

Then call it with, e.g:
`:Fast '(send {(const nil :FactoryBot) nil} {:create :build} (sym :user)' spec/**/*.rb`

**Comment by nerdrew** [comment]
It would simplify the integration a little bit (simpler error format) if there was an `--emacs` or `--vim` (`rg` has `--vimgrep`, `rubocop` has `--format emacs`) output mode. I think both do the same thing: print the match on a single line with line and column number. E.g.:

`path/to/file.rb:1234:20 my_method(:boom, arg1: 1)`

I'm not sure how complicated combining the matched code onto one line would be though.

BTW: This tool rocks. Thanks!

**Comment by nerdrew** [comment]
LOL. Just saw this too: https://no-color.org/ Maybe we should check that ENV var too.

**Comment by jonatas** [comment]
Looks like Ruby still does not support no-color. 

For building the extra formatting is quite easy. Basically `Fast.report` can be supporting multiple outputs and we can delegate it to the different implementation. We actually support `--ast`, `--no-color` and `--headless`. If we introduce `--format` we should consider having the other options combined with the same purpose.
Thanks for sharing your Vim function. I'll probably make it part of our official docs and make it easy for other people to set up their Fast function on vim too :)

I'll start using your snippets in my local vim.

**Comment by jonatas** [comment]
After a few hours on it, my solution was simply open another terminal with `vterm` and the result is quite good:

```vim
nnoremap <Leader>ff :vsplit \| terminal fast "()" % <Left><Left><Left><Left><Left>
```

and then with `<space>ff` I can jump into a search in the current file with focus in the :

Screen Shot 2019-11-21 at 00 21 05
 
The funny part is that the colors worked well for me and copy and paste also worked well with colors.

I still need to facilitate the "GOTO" part implementing more details to allow to go and automatically focus on the file and so on.

**Comment by jonatas** [comment]
Added a small topic here to not lose the track: https://jonatas.github.io/fast/editors-integration/

**Pull request merged**